### PR TITLE
[IOPAY-148] Configure pipelines for io-pay-portal-fe and io-functions-pay-portal

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of changes
- Add deploy and code-review pipelines for [io-pay-portal-fe](https://github.com/pagopa/io-pay-portal/tree/main/io-pay-portal-fe);
- Add deploy and code-review pipelines for [io-functions-pay-portal](https://github.com/pagopa/io-pay-portal/tree/main/io-functions-pay-portal);

### CODE REVIEW

**Review by uolter — APPROVED** [review]

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 1]
```diff
@@ -0,0 +1,161 @@
+variable "io-functions-pay-portal" {
+  default = {
+    repository = {
+      organization   = "pagopa"
+      name           = "io-functions-pay-portal"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:5]:
```suggestion
      name           = "io-pay-portal"
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      branch_name    = "main"
+      pipelines_path = ".devops"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:7]:
```suggestion
      pipelines_path = "."
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        include = []
+      }
+    }
+  }
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-functions-pay-portal.repository.organization}/${var.io-functions-pay-portal.repository.name}"
+    branch_name           = var.io-functions-pay-portal.repository.branch_name
+    yml_path              = "${var.io-functions-pay-portal.repository.pipelines_path}/io-functions-pay-portal/code-review-pipelines.yml"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:47]:
```suggestion
    yml_path              = "${var.io-functions-pay-portal.repository.pipelines_path}/io-functions-pay-portal/.devops/code-review-pipelines.yml"
```
https://github.com/pagopa/io-pay-portal/tree/main/io-functions-pay-portal/.devops
you need to rename code review file

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  }
+}
+
+# code review
+resource "azuredevops_build_definition" "io-functions-pay-portal-code-review" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-pay-portal.repository.name}.code-review"
+  path       = "\\${var.io-functions-pay-portal.repository.name}/io-functions-pay-portal"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:23]:
```suggestion
  path       = "\\${var.io-functions-pay-portal.repository.name}\io-functions-pay-portal"
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  }
+}
+
+# code review
+resource "azuredevops_build_definition" "io-functions-pay-portal-code-review" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-pay-portal.repository.name}.code-review"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:22]:
```suggestion
  name       = "io-functions-pay-portal.code-review"
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  authorized    = true
+  type          = "endpoint"
+}
+
+# deploy
+resource "azuredevops_build_definition" "io-functions-pay-portal-deploy" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-rw, azuredevops_serviceendpoint_azurerm.PROD-IO, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-pay-portal.repository.name}.deploy"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:84]:
```suggestion
  name       = "io-functions-pay-portal.deploy"
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  path       = "\\${var.io-functions-pay-portal.repository.name}"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:85]:
```suggestion
  path       = "\\${var.io-functions-pay-portal.repository.name}\io-functions-pay-portal"
```

File: azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-functions-pay-portal.repository.organization}/${var.io-functions-pay-portal.repository.name}"
+    branch_name           = var.io-functions-pay-portal.repository.branch_name
+    yml_path              = "${var.io-functions-pay-portal.repository.pipelines_path}/io-functions-pay-portal/deploy-pipelines.yml"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-functions-pay-portal.tf:91]:
```suggestion
    yml_path              = "${var.io-functions-pay-portal.repository.pipelines_path}/io-functions-pay-portal/.devops/deploy-pipelines.yml"
```
https://github.com/pagopa/io-pay-portal/tree/main/io-functions-pay-portal/.devops
you need to rename deploy file

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
thanks @infantesimone for your pr, here we need to set gitgub repo name.
PS: I see that this repo has both backend and frontend, so we need some changes

also we need to change pagopa user as here https://github.com/pagopa/gitops/pull/22#issuecomment-821310752

File: azure-devops/projects/io-pay-projects/io-pay-portal-fe.tf [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+resource "azuredevops_resource_authorization" "io-pay-portal-fe-deploy-azurerm-PROD-IO-auth" {
+  depends_on = [azuredevops_serviceendpoint_azurerm.PROD-IO, azuredevops_build_definition.io-pay-portal-fe-deploy, time_sleep.wait]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_azurerm.PROD-IO.id
+  definition_id = azuredevops_build_definition.io-pay-portal-fe-deploy.id
+  authorized    = true
+  type          = "endpoint"
+}
```
Comment (pasqualedevita) [inline azure-devops/projects/io-pay-projects/io-pay-portal-fe.tf:214]:
```suggestion
}

resource "azurerm_role_assignment" "io-pay-deploy-azurerm-PROD-IO-iopstcdniopayportal" {
  depends_on = [data.azuread_service_principal.service_principals]

  principal_id         = data.azuread_service_principal.service_principals[local.PROD-IO-UID].id
  role_definition_name = "Storage Blob Data Contributor"
  scope                = "/subscriptions/${data.azurerm_key_vault_secret.key_vault_secret["TTDIO-PROD-IO-SUBSCRIPTION-ID"].value}/resourceGroups/${var.io-pay-portal-fe.pipeline.resource_group_azure}/providers/Microsoft.Storage/storageAccounts/${var.io-pay-portal-fe.pipeline.production_storage_account_name}"
}
```

Add permission to service principal to write into production_storage_account_name

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — DISMISSED** [review]
LGTM

**Comment by pasqualedevita** [comment]
@infantesimone can u change pipeline names here? https://github.com/pagopa/io-pay-portal/tree/main/io-functions-pay-portal/.devops

**Comment by GiovanniMancini** [comment]
@pasqualedevita @uolter  We need to change a couple of environment variables in this PR. We should merge it and open another PR or do you prefer a commit on this branch?
cc: @infantesimone

**Comment by pasqualedevita** [comment]
> @pasqualedevita @uolter We need to change a couple of environment variables in this PR. We should merge it and open another PR or do you prefer a commit on this branch?
> cc: @infantesimone

use this pr

**Review by pasqualedevita — APPROVED** [review]
Ok, project migrated into new pagopaspa devops org (https://dev.azure.com/pagopaspa/io-pay-projects)

**Comment by pasqualedevita** [comment]
@infantesimone @GiovanniMancini tomorrow can u test that it's all ok for u? login here https://dev.azure.com/pagopaspa/io-pay-projects

**Comment by infantesimone** [comment]
All pipelines are working properly, thanks @pasqualedevita

**Pull request merged**